dbt_coves/tasks/extract/airbyte.py

```diff
@@ -0,0 +1,212 @@
+        sources_path = path / "sources"
+        sources_path.mkdir(parents=True, exist_ok=True)
+        destinations_path = path / "destinations"
+        destinations_path.mkdir(parents=True, exist_ok=True)
+
+        self.connections_extract_destination = os.path.abspath(connections_path)
+        self.destinations_extract_destination = os.path.abspath(destinations_path)
+        self.sources_extract_destination = os.path.abspath(sources_path)
+
+        self.airbyte_api_caller = AirbyteApiCaller(airbyte_host, airbyte_port)
+
+        console.print(
+            f"Extracting Airbyte's [b]Source[/b], [b]Destination[/b] and [b]Connection[/b] configurations to {os.path.abspath(path)}\n"
+        )
+
+        dbt_sources_list = shell.run_dbt_ls(
+            "dbt ls --resource-type source",
+            None,
+        )
+        if dbt_sources_list:
+            dbt_sources_list = self._remove_airbyte_prefix(dbt_sources_list)
+            for source in dbt_sources_list:
+                # Obtain db.schema.table
+                source_db, source_schema, source_table = [
+                    element.lower() for element in source.split(".")
+                ]
+
+                source_connection = self._get_airbyte_connection_for_table(source_table)
+
+                if source_connection:
+                    source_destination = self._get_airbyte_destination_from_id(
+                        source_connection["destinationId"]
+                    )
+                    source_source = self._get_airbyte_source_from_id(
+                        source_connection["sourceId"]
+                    )
+
+                    if source_destination and source_source:
+                        self._save_json_connection(source_connection)
+                        self._save_json_destination(source_destination)
+                        self._save_json_source(source_source)
+                else:
+                    print(f"There is no Airbyte Connection for source: {source}")
+            console.print(
+                f"""Extraction successful!
+[u]Sources[/u]: {self.extraction_results['sources']}
+[u]Destinations[/u]: {self.extraction_results['destinations']}
+[u]Connections[/u]: {self.extraction_results['connections']}
+"""
+            )
+            return 0
+        else:
+            raise AirbyteExtractorException("There are no DBT Sources compiled")
+
+    def _remove_airbyte_prefix(self, sources_list):
+        return [
+            source.lower().replace("_AIRBYTE_RAW_".lower(), "")
+            for source in sources_list
+        ]
+
+    def _get_airbyte_connection_for_table(self, table):
+        """
+        Given a table name, returns the corresponding airbyte connection
+        """
+        for conn in self.airbyte_api_caller.airbyte_connections_list:
+            for stream in conn["syncCatalog"]["streams"]:
+                if stream["stream"]["name"].lower() == table:
+                    return conn
```

Comment:
Missing `raise` after `for` loop

Reply:
Added:

`raise AirbyteExtractorException(
            f"Airbyte extract error: there are no connections for table {table}"
        )`

Reply:
7f7dcc2[](url)